apps/src/aichat/views/TeacherFeedbackFooter.tsx

```diff
@@ -0,0 +1,113 @@
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+
+import aichatI18n from '../locale';
+import {submitTeacherFeedback} from '../redux/aichatRedux';
+import {ChatMessage, TeacherFeedback} from '../types';
+
+import moduleStyles from './teacher-feedback-footer.module.scss';
+
+interface Props {
+  isProfanityViolation: boolean;
+  chatMessage: ChatMessage;
+}
+
+const TeacherFeedbackFooter: React.FC<Props> = ({
+  isProfanityViolation,
+  chatMessage,
+}) => {
+  const dispatch = useAppDispatch();
+  const [thumbsUp, setThumbsUp] = useState(
+    chatMessage.teacherFeedback === TeacherFeedback.PROFANITY_AGREE
+  );
+  const [thumbsDown, setThumbsDown] = useState(
+    chatMessage.teacherFeedback === TeacherFeedback.PROFANITY_DISAGREE
+  );
+  const handleThumbClick = (thumbsUp: boolean, thumbsDown: boolean) => {
+    setThumbsUp(thumbsUp);
+    setThumbsDown(thumbsDown);
+    const teacherFeedback = thumbsUp
+      ? TeacherFeedback.PROFANITY_AGREE
+      : thumbsDown
+      ? TeacherFeedback.PROFANITY_DISAGREE
+      : undefined;
+    dispatch(submitTeacherFeedback({...chatMessage, teacherFeedback}));
+  };
+
+  const [flaggedAsInappropriate, setFlaggedAsInappropriate] = useState(
+    chatMessage.teacherFeedback === TeacherFeedback.CLEAN_DISAGREE
+  );
+  const handleFlagClick = (toggle: boolean) => {
+    setFlaggedAsInappropriate(toggle);
+    const teacherFeedback = toggle ? TeacherFeedback.CLEAN_DISAGREE : undefined;
+    dispatch(submitTeacherFeedback({...chatMessage, teacherFeedback}));
+  };
+
+  return (
+    <>
+      {isProfanityViolation && (
+        <div className={moduleStyles.teacherFeedbackContainer}>
+          <EmText>Was this content flagged correctly?</EmText>
+          <Button
+            color={buttonColors.black}
+            disabled={false}
+            icon={{iconName: 'thumbs-up', iconStyle: 'solid'}}
+            isIconOnly={true}
+            onClick={() => handleThumbClick(!thumbsUp, false)}
+            size="xs"
+            type={thumbsUp ? 'primary' : 'tertiary'}
+          />
+          <Button
+            color={buttonColors.black}
+            disabled={false}
+            icon={{iconName: 'thumbs-down', iconStyle: 'solid'}}
+            isIconOnly={true}
+            onClick={() => handleThumbClick(false, !thumbsDown)}
+            size="xs"
+            type={thumbsDown ? 'primary' : 'tertiary'}
+          />
+        </div>
+      )}
+      {!isProfanityViolation && (
+        <div
+          className={classNames(
+            moduleStyles.teacherFeedbackContainer,
+            chatMessage.role === Role.ASSISTANT &&
+              moduleStyles.assistanctFeedbackContainerOverride
+          )}
+        >
+          <EmText
+            className={classNames(
+              moduleStyles.hiddenTilHover,
+              flaggedAsInappropriate && moduleStyles.showAlways
+            )}
+          >
+            {flaggedAsInappropriate
+              ? aichatI18n.chatMessage_hasBeenFlagged()
```

Comment:
initially, the flag is shown and if a user hovers on it, there is a message 'Flag message as inappropriate'. If a teacher flags a message, and then changes their mind, they can click on the flag again to unflag. But I wonder if we should provide a visual indication that they can unflag the message.

![Screenshot 2024-12-20 at 2 44 08 PM](https://github.com/user-attachments/assets/67942857-5221-45e1-ae4a-63b40fcf3d82)

Reply:
There is a hover state change when the button is flagged, when you mouse on and off of it. Is that what you mean? They do have to mouse off of it after clicking to see that animation.

Reply:
Following up from offline convo - makes sense now. When it's toggled off and you hover, it tells you what will happen if you click it. When it's toggled on, it tells you the current state (has been flagged) vs what will happen when you click it (unflag).

Reply:
Noted in the follow-up work!

Reply:
Noted in follow-ups